### PR TITLE
feat: add issue udt view

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: apps/ckit-app/src/components/Issue/index.tsx [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+interface ModalFormProps {
+  visible: boolean;
+  setVisible: (visible: boolean) => void;
+  operationKind: OperationKind;
+}
+
+interface ModalFormValues {
+  recipient: Address;
+  amount: HexString;
```
Comment (homura) [inline apps/ckit-app/src/components/Issue/index.tsx]:
```suggestion
  amount: HexNumber;
```

File: apps/ckit-app/src/components/Issue/index.tsx [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+}
+
+interface ModalFormErrors {
+  recipient?: Address;
+  amount?: HexString;
```
Comment (homura) [inline apps/ckit-app/src/components/Issue/index.tsx]:
```suggestion
  amount?: HexNumber;
```

File: apps/ckit-app/src/hooks/useConfigStorage.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+export interface LocalConfig {
+  ckitConfig: CkitConfig;
+  mecuryRPC: string;
+  ckbRPC: string;
+  nervosExploreTxUrlPrefix: string;
+  nervosExploreAddressUrlPrefix: string;
+}
+
+export function useConfigStorage(): [LocalConfig, (newValue: LocalConfig) => void, () => void] {
```
Comment (homura) [inline apps/ckit-app/src/hooks/useConfigStorage.tsx:16]:
```suggestion
// TODO remove random when predefined config is ready
export function useConfigStorage(): [LocalConfig, (newValue: LocalConfig) => void, () => void] {
```

File: apps/ckit-app/src/hooks/useSendIssueTx.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+export function useSendIssueTx(): UseMutationResult<{ txHash: Hash }, unknown, SendIssueTxInput> {
+  const { selectedWallet } = WalletContainer.useContainer();
+  const ckitProvider = CkitProviderContainer.useContainer();
+  const [localConfig] = useConfigStorage();
+
+  return useMutation(
+    ['sendIssueTx'],
+    async (input: SendIssueTxInput) => {
+      if (!selectedWallet?.signer) throw new Error('exception: signer not set');
```
Comment (homura) [inline apps/ckit-app/src/hooks/useSendIssueTx.tsx]:
```suggestion
      if (!selectedWallet?.signer) throw new Error('Signer could not be found, the wallet may be disconnected, you can try to reconnect');
```
is the error message for user? if so we can change it to the above. I think a good error message to end user should be cause + solution

File: apps/ckit-app/src/hooks/useSendIssueTx.tsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+import { useMutation, UseMutationResult } from 'react-query';
+import { useConfigStorage } from './useConfigStorage';
+import { CkitProviderContainer, WalletContainer } from 'containers';
+import { hasProp } from 'utils';
+
+export interface SendIssueTxInput {
+  recipient: Address;
+  amount: HexNumber;
+  operationKind: 'invite' | 'issue';
+  setModalVisible: (visible: boolean) => void;
```
Comment (homura) [inline apps/ckit-app/src/hooks/useSendIssueTx.tsx]: an UI business should NOT in a hooks, this leads to coupling of UI and business logic
```suggestion
```

we can just

```ts
const {mutate: sendTx} = useSendIssueTx()

onSubmit: () => {
  return sendTx().then(
    () => // do something on success,
    () => do something on error,
  )
}
```

File: apps/ckit-app/src/components/Issue/index.tsx [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+                  <label htmlFor="recipient">{recipientFieldDisplayName}</label>
+                </Col>
+                <Col span={16}>
+                  <Field name="recipient" type="text" placeholder="ckb address" />
+                </Col>
+              </Row>
+              <ErrorMessage
+                name="recipient"
+                children={(errorMessage) => {
+                  return <Typography.Text type={'danger'}>{errorMessage}</Typography.Text>;
```
Comment (homura) [inline apps/ckit-app/src/components/Issue/index.tsx]:
```suggestion
                  return <Typography.Text type="danger">{errorMessage}</Typography.Text>;
```

File: apps/ckit-app/src/components/Issue/index.tsx [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+                    <label htmlFor="amount">amount:</label>
+                  </Col>
+                  <Col span={16}>
+                    <Field name="amount" type="text" />
+                  </Col>
+                </Row>
+                <ErrorMessage
+                  name="amount"
+                  children={(errorMessage) => {
+                    return <Typography.Text type={'danger'}>{errorMessage}</Typography.Text>;
```
Comment (homura) [inline apps/ckit-app/src/components/Issue/index.tsx]:
```suggestion
                    return <Typography.Text type="danger">{errorMessage}</Typography.Text>;
```

Comment (ghost) [reply apps/ckit-app/src/components/Issue/index.tsx]: fixed

File: apps/ckit-app/src/components/Issue/index.tsx (hunk 6 above)
Comment (ghost) [reply apps/ckit-app/src/components/Issue/index.tsx]: fixed

File: apps/ckit-app/src/hooks/useSendIssueTx.tsx (hunk 5 above)
Comment (ghost) [reply apps/ckit-app/src/hooks/useSendIssueTx.tsx]: fixed

File: apps/ckit-app/src/hooks/useSendIssueTx.tsx (hunk 4 above)
Comment (ghost) [reply apps/ckit-app/src/hooks/useSendIssueTx.tsx]: fixed

File: apps/ckit-app/src/hooks/useConfigStorage.tsx (hunk 3 above)
Comment (ghost) [reply apps/ckit-app/src/hooks/useConfigStorage.tsx:16]: fixed

File: apps/ckit-app/src/components/Issue/index.tsx (hunk 2 above)
Comment (ghost) [reply apps/ckit-app/src/components/Issue/index.tsx]: fixed

File: apps/ckit-app/src/components/Issue/index.tsx (hunk 1 above)
Comment (ghost) [reply apps/ckit-app/src/components/Issue/index.tsx]: fixed

**Pull request merged**